src/main/java/org/hisp/dhis/antlr/AntlrParserUtils.java

```diff
@@ -188,106 +188,132 @@ public static Date castDate( Object object )
      */
     public static Object castClass( Class<?> clazz, Object object )
     {
-        if ( object instanceof Double )
+        if ( object == null )
         {
-            return castDoubleObject( (Double) object, clazz );
+            return null;
         }
 
-        if ( object instanceof String )
+        if ( clazz == Double.class && makeDouble( object ) != null )
```

Comment:
```suggestion
        Object result;
        if ( clazz == Double.class && ( result = makeDouble( object )) != null )
```

Reply:
I am suggesting a pretty ugly change but that could reduce quite a bit the computation done to cast types.
I think this can be important as this method can be called a lot of times during one evaluation

If we make this change, we also need to change the other parts of the method

Reply:
Thanks, I like the idea, so much that I've done it also in `AntlrOperatorCompare`. I also felt a little funny about calling the conversion methods twice instead of once. I'd also thought I could assign the converted value to a variable within the if statement and then test for null, but I'd never done that in Java before. (I did it many times in C.) Then I forgot to try it out. So thanks for the idea! See my new commit.